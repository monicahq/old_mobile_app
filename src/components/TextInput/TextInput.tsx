import React, {PureComponent} from 'react';
import {Text, TextInput as RNTextInput, View} from 'react-native';
import {borderColor, styles} from './TextInput.styles';

interface ITextInputProps {
  id?: string;
  title: string;
  touched?: boolean;
  error?: string;
}

export class TextInput extends PureComponent<ITextInputProps> {
  public static defaultProps = {
    touched: false,
    error: null,
  };
  private textInput;

  public focus() {
    this.textInput.focus();
  }

  public textInputRef = ref => (this.textInput = ref);

  public render() {
    const {title, error, touched, id, ...props} = this.props;

    return [
      // TITLE + ERROR
      <View key={0} style={styles.title}>
        {!!title && <Text>{title}</Text>}
        {!!error &&
          touched && (
            <View style={styles.flexRow}>
              <Text> - </Text>
              <Text style={styles.errorText}>{error}</Text>
            </View>
          )}
      </View>,
      // TEXTINPUT
      <RNTextInput
        testID={id}
        ref={this.textInputRef}
        key={1}
        style={styles.textInput}
        autoCorrect={false}
        autoCapitalize="none"
        underlineColorAndroid={borderColor}
        {...props}
      />,
    ];
  }
}
