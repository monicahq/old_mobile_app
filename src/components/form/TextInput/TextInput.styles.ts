import {PixelRatio, Platform, StyleSheet} from 'react-native';

export const borderColor = '#b5b5b5';

export const styles = StyleSheet.create({
  flexRow: {
    flexDirection: 'row',
  },
  title: {
    width: '100%',
    flexDirection: 'row',
  },
  errorText: {
    width: '100%',
    color: 'red',
  },
  textInput: {
    height: 40,
    width: '100%',
    borderColor,
    borderBottomWidth: Platform.OS === 'ios' ? 1 / PixelRatio.get() : 0,
    marginBottom: 20,
  },
  titleMultine: {
    fontSize: 11,
  },
});
