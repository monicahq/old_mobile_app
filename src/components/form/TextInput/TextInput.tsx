import React, {PureComponent} from 'react';
import {
  Text,
  TextInput as RNTextInput,
  TextInputProps,
  View,
} from 'react-native';
import {borderColor, styles} from './TextInput.styles';

interface ITextInputProps extends TextInputProps {
  id?: string;
  title: string;
  touched?: boolean;
  error?: string;
  onSubmitEditing?: (...args) => void;
}

export class TextInput extends PureComponent<ITextInputProps> {
  public static defaultProps = {
    touched: false,
    error: null,
  };
  private textInput;

  public focus() {
    this.textInput.focus();
  }

  public textInputRef = ref => (this.textInput = ref);

  public render() {
    const {title, error, touched, id, ...props} = this.props;

    const multilineTextStyle = props.multiline ? styles.titleMultine : null;

    return [
      // TITLE + ERROR
      <View key={0} style={styles.title}>
        {!!title && <Text style={multilineTextStyle}>{title}</Text>}
        {!!error &&
          touched && (
            <View style={styles.flexRow}>
              <Text style={multilineTextStyle}> - </Text>
              <Text
                style={[styles.errorText, multilineTextStyle]}
                testID={`input-error-${id}`}
              >
                {error}
              </Text>
            </View>
          )}
      </View>,
      // TEXTINPUT
      <RNTextInput
        testID={`input-${id}`}
        ref={this.textInputRef}
        key={1}
        style={styles.textInput}
        autoCorrect={false}
        autoCapitalize="none"
        underlineColorAndroid={borderColor}
        {...props}
      />,
    ];
  }
}
