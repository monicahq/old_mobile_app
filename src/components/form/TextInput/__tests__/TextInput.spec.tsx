import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {TextInput} from '../TextInput';

describe('Components', () => {
  describe('TextInput', () => {
    it('ios should renders correctly', () => {
      expect(toJson(shallow(<TextInput title="My title" />))).toMatchSnapshot();
      expect(
        toJson(shallow(<TextInput title="My title" touched={true} />))
      ).toMatchSnapshot();

      expect(
        toJson(
          shallow(
            <TextInput title="My title" touched={true} error="my-error" />
          )
        )
      ).toMatchSnapshot();

      expect(
        toJson(shallow(<TextInput title="My title" error="My errors" />))
      ).toMatchSnapshot();
    });

    it('android should renders correctly', () => {
      jest.mock('Platform', () => ({
        OS: 'android',
      }));

      expect(toJson(shallow(<TextInput title="My title" />))).toMatchSnapshot();
      expect(
        toJson(shallow(<TextInput title="My title" touched={true} />))
      ).toMatchSnapshot();
    });

    it('should handle focus event', () => {
      const tree = shallow(<TextInput title="My title" />);

      tree.instance().textInput = {
        focus: jest.fn(),
      };
      tree.instance().focus();

      expect(tree.instance().textInput.focus.mock.calls.length).toBe(1);
      expect(tree.instance().textInput.focus.mock.calls[0]).toEqual([]);
    });

    it('should initialize textInput ref', () => {
      const textInputVal = 'test-text-input';

      const tree = shallow(<TextInput title="My title" />);
      tree.instance().textInputRef(textInputVal);

      expect(tree.instance().textInput).toBe(textInputVal);
    });
  });
});
