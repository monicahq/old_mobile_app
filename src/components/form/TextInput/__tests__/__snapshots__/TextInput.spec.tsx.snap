// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components TextInput android should renders correctly 1`] = `
Array [
  <Component
    key="0"
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  />,
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    key="1"
    style={
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#b5b5b5",
        "height": 40,
        "marginBottom": 20,
        "width": "100%",
      }
    }
    testID="input-undefined"
    underlineColorAndroid="#b5b5b5"
  />,
]
`;

exports[`Components TextInput android should renders correctly 2`] = `
Array [
  <Component
    key="0"
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  />,
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    key="1"
    style={
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#b5b5b5",
        "height": 40,
        "marginBottom": 20,
        "width": "100%",
      }
    }
    testID="input-undefined"
    underlineColorAndroid="#b5b5b5"
  />,
]
`;

exports[`Components TextInput ios should renders correctly 1`] = `
Array [
  <Component
    key="0"
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  />,
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    key="1"
    style={
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#b5b5b5",
        "height": 40,
        "marginBottom": 20,
        "width": "100%",
      }
    }
    testID="input-undefined"
    underlineColorAndroid="#b5b5b5"
  />,
]
`;

exports[`Components TextInput ios should renders correctly 2`] = `
Array [
  <Component
    key="0"
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  />,
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    key="1"
    style={
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#b5b5b5",
        "height": 40,
        "marginBottom": 20,
        "width": "100%",
      }
    }
    testID="input-undefined"
    underlineColorAndroid="#b5b5b5"
  />,
]
`;

exports[`Components TextInput ios should renders correctly 3`] = `
Array [
  <Component
    key="0"
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  />,
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    key="1"
    style={
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#b5b5b5",
        "height": 40,
        "marginBottom": 20,
        "width": "100%",
      }
    }
    testID="input-undefined"
    underlineColorAndroid="#b5b5b5"
  />,
]
`;

exports[`Components TextInput ios should renders correctly 4`] = `
Array [
  <Component
    key="0"
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  />,
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    key="1"
    style={
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#b5b5b5",
        "height": 40,
        "marginBottom": 20,
        "width": "100%",
      }
    }
    testID="input-undefined"
    underlineColorAndroid="#b5b5b5"
  />,
]
`;
